src/cli.cc

```diff
@@ -96,6 +109,30 @@ ExitCode CompleteInstall(AkliteClient& client) {
       exit_code = ExitCode::InstallNeedsRebootForBootFw;
       break;
     }
+    case InstallResult::Status::Failed: {
+      // do rollback
+      // check rollback type, bootloader (boot on the previous version) or App driven
+      const auto current{client.GetCurrent()};  // returns Target a device is booted on
+      if (current.Sha256Hash() != target.Sha256Hash()) {
+        // ostree rollback
+        auto rollback_installer = client.CheckAppsInSync();
+        if (rollback_installer) {
+          auto rollback_download_res = rollback_installer->Download();
+          // TODO
+          auto rollback_install_res = rollback_installer->Install();
+          if (rollback_install_res.status == InstallResult::Status::Ok) {
+            exit_code = ExitCode::InstallRollbackOk;
+          } else {
+            exit_code = ExitCode::InstallRollbackFailed;
```

Comment:
Yet another "hard" question:
- what is the user expected to do in case if both installation and rollback fail?